pandas/core/indexing.py

```diff
@@ -1130,6 +1144,17 @@ def __getitem__(self, key):
             axis = self.axis or 0
 
             maybe_callable = com.apply_if_callable(key, self.obj)
+            if (
+                self.name == "iloc"
+                and callable(key)
+                and isinstance(maybe_callable, tuple)
+            ):
+                warnings.warn(
+                    "Returning a tuple from a callable in iLocation indexing "
+                    "is deprecated and will be removed in a future version",
+                    FutureWarning,
+                    stacklevel=find_stack_level(),
+                )
```

Comment:
This is very similar to the above. Can you refactor them into a method function `apply_to_iloc_if_callable` or similar.

Also the same issue is with `.loc` indexing, isn't it?

Reply:
Also, you should be able to return a tuple, if the index is a `MultiIndex` though.

Reply:
> Also the same issue is with .loc indexing, isn't it?

No. A singleton tuple is (potentially) a valid key for `.loc` indexing (although this is also treated inconsistently and generally incorrectly by this code, see https://github.com/pandas-dev/pandas/issues/53535).

Reply:
Done (the refactor).